src/backend/distributed/connection/connection_management.c

```diff
@@ -1303,7 +1306,10 @@ ShouldShutdownConnection(MultiConnection *connection, const int cachedConnection
 		   cachedConnectionCount >= MaxCachedConnectionsPerWorker ||
 		   connection->forceCloseAtTransactionEnd ||
 		   PQstatus(connection->pgConn) != CONNECTION_OK ||
-		   !RemoteTransactionIdle(connection);
+		   !RemoteTransactionIdle(connection) ||
+		   (MaxCachedConnectionLifetime >= 0 &&
+			TimestampDifferenceExceeds(connection->connectionStart, GetCurrentTimestamp(),
```

Comment:
As this function can be called extensively, I tried to check if calling `GetCurrentTimestamp` would  add any overhead. My quick tests show that, it doesn't. If we observe anything like this in the future, we might consider using a cached value